src/betterbib/tools.py

```diff
@@ -442,3 +448,38 @@ def filter_fields(data, excludes=None):
         if entry.fields:
             entry.fields = {k: v for k, v in entry.fields.items() if k not in excludes}
     return data
+
+
+def bibtex_parser(infile):
+    """
+    Returns the parsed bibtex data and adds context to the exception
+
+        Parameters:
+            infile (FileType("r")): file to be parsed
+
+        Returns:
+            data (dict): bibtex entries
+    """
+    try:
+        data = bibtex.Parser().parse_file(infile)
+        return data
+
+    except Exception as e:
+        print("There was an error when parsing " + infile.name)
+        raise e
+
+
+def bibtex_writer(string: str, outfile, to_file: bool = False):
```

Comment:
I don't like the arguments of `bibtex_writer`. If `to_file` is false, the `outfile` argument isn't used at all. Better `outfile=None` as default. (Or something along this line.)

Reply:
The function can also get a more generic name, like `write`.